### ISSUE TITLE
Unable to extend Error when targeting es5 with TS 2.1

### ISSUE THREAD

**Issue by asvetliakov** [Microsoft/TypeScript#12058]
**TypeScript Version:**  2.1.0-dev.20161104

**Code**
tsconfig.json
```
{                       
  "compilerOptions": {  
    "target": "es5",    
    "module": "commonjs"
  }      
}                       
```

```ts
// A *self-contained* demonstration of the problem follows...
class Test extends Error {
        constructor(message: string) {
                super(message);
        }
}

const test = new Test("test");

console.log(`Instance of error: ${test instanceof Error}`);
console.log(`Instance of test: ${test instanceof Test}`);
```

**Expected behavior:**
```
Instance of error: true
Instance of test: true
```

**Actual behavior:** 
```
Instance of error: true
Instance of test: false
```

It works with TS 2.0.7

**Comment by aluanhaddad** [comment]
For reference, this affects all code that extends a function that is both callable constructable. For example, see #11919. Since `Error(msg)` is a factory, its return value is propagated through the super call and used instead of `this`.

**Comment by asvetliakov** [comment]
I won't complain if it's intentional change (workaround-ed easily), but it at least it should be documented (i.e. breaking change) or even better the TS compiler should give error on es5 target when 2.1 will be released.

**Comment by aluanhaddad** [comment]
@asvetliakov I agree. I don't know what the intent exactly is. Either way subclassing builtins like `Array` and `Error` is broken. An error should be issued.

**Comment by DanielRosenwasser** [comment]
It actually is an intentional change [and documented here](https://github.com/Microsoft/TypeScript-wiki/blob/master/Breaking-Changes.md#extending-built-ins-like-error-array-and-map-may-no-longer-work). I apologize about the inconvenience.

**Comment by DanielRosenwasser** [comment]
Marking as a duplicate of #12123.
